File: docs/WidgetMill/packages/spec/src/nodes/divider.ts
import { z } from "zod";
import { BindableColor } from "../scalars";

/** Divider — linha divisória horizontal (catálogo Flutter). */
export const DividerNode = z.object({
  type: z.literal("divider"),
  props: z
    .object({
      height: z.number().optional(),
      thickness: z.number().optional(),
      indent: z.number().optional(),
      endIndent: z.number().optional(),
      color: BindableColor.optional(),
    })
    .default({}),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/divider.dart
import 'package:flutter/material.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Divider — linha divisória horizontal.
Widget buildDivider(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Divider(
    height: parseDouble(p['height']),
    thickness: parseDouble(p['thickness']),
    indent: parseDouble(p['indent']),
    endIndent: parseDouble(p['endIndent']),
    color: parseColor(p['color']),
  );
}
